client/js/widgets/widget.js

```diff
@@ -1268,11 +1261,13 @@ export class Widget extends StateManaged {
         setDefaults(a, { collection: 'DEFAULT' });
         const collection = getCollection(a.collection);
         if(collection) {
+          const promises = [];
           for(const w of collections[collection]) {
-            await removeWidgetLocal(w.get('id'));
+            promises.push(removeWidgetLocal(w.get('id')));
             for(const c in collections)
               collections[c] = collections[c].filter(x=>x!=w);
           }
+          await Promise.all(promises);
```

Comment:
This change seems unrelated to the PR. Might be an improvement but I don't see a reason to have it in here.